### PR TITLE
Add stream service

### PR DESCRIPTION
**Pull request by irlirla**
-Add solution with name "LT.DigitalOffice.StreamService";
-add directory with name "Solution items" and put the following files there: ".dockerignore", ".editorconfig", ".gitignore", "Common.Configurations.targets", "Dockerfile", "main.yml", "README.md"
-add directory with name "src" and put the following items:
application wit name "StreamService" (add appsettings.json, set Program.cs, Startup.cs), add following DLLs: StreamService.Broker (add "Consumers" directory), StreamService.Business (add "Commands" directory), StreamService.Data (add "Interfaces" directory), StreamService.Data.Provider (add "IDataProvider" interface, StreamService.Data.Provider.MsSql.Ef (add "Migrations" directory and "StreamServiceDbContext" class, StreamService.Mappers, StreamService.Models.Db, StreamService.Models.Dto, StreamService.Validation

### CODE REVIEW

**Review by Cheephot — APPROVED** [review]

File: src/StreamService/StreamService.csproj [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
```
Comment (ekaterinaryzhenkova) [inline src/StreamService/StreamService.csproj:2]: maybe you need a connection with CommonConfigurationTargets here and in libraries, consult Mariya

File: src/StreamService/Properties/launchSettings.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿{
```
Comment (ekaterinaryzhenkova) [inline src/StreamService/Properties/launchSettings.json:1]: are you sure this file should look like this?

File: src/StreamService/Program.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.StreamService
+{
+    public class Program
```
Comment (ekaterinaryzhenkova) [inline src/StreamService/Program.cs]: are you sure this class should look like this?

File: StreamService.Broker.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+﻿using System;
+
+public class Class1
```
Comment (ekaterinaryzhenkova) [inline StreamService.Broker.cs]: do you need this class?

File: src/StreamService/Startup.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.StreamService
+{
+    public class Startup : BaseApiInfo
```
Comment (ekaterinaryzhenkova) [inline src/StreamService/Startup.cs]: are you sure this class should look like this?

File: src/StreamService/appsettings.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+  "Logging": {
```
Comment (ekaterinaryzhenkova) [inline src/StreamService/appsettings.json]: are you sure this file should look like this?

File: src/StreamService/appsettings.Development.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
+  "Logging": {
```
Comment (ekaterinaryzhenkova) [inline src/StreamService/appsettings.Development.json]: are you sure this file should look like this?

File: src/StreamService/Properties/launchSettings.json (hunk 2 above)
Comment (irlirla) [reply src/StreamService/Properties/launchSettings.json:1]: fixed

File: src/StreamService/Program.cs (hunk 3 above)
Comment (irlirla) [reply src/StreamService/Program.cs]: fixed

File: StreamService.Broker.cs (hunk 4 above)
Comment (irlirla) [reply StreamService.Broker.cs]: fixed

File: src/StreamService/Startup.cs (hunk 5 above)
Comment (irlirla) [reply src/StreamService/Startup.cs]: fixed

File: src/StreamService/appsettings.Development.json (hunk 7 above)
Comment (irlirla) [reply src/StreamService/appsettings.Development.json]: fixed

File: src/StreamService/Program.cs [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+          }
+          finally
+          {
+            Log.CloseAndFlush();
+          }
+      }
+
+      public static IHostBuilder CreateHostBuilder(string[] args) =>
+          Host.CreateDefaultBuilder(args)
+              .UseSerilog()
```
Comment (MariyaSin) [inline src/StreamService/Program.cs]: check tabs

File: src/StreamService/StreamService.csproj (hunk 1 above)
Comment (irlirla) [reply src/StreamService/StreamService.csproj:2]: fixed

File: src/StreamService/appsettings.json (hunk 6 above)
Comment (irlirla) [reply src/StreamService/appsettings.json]: fixed

File: src/StreamService/Program.cs (hunk 8 above)
Comment (irlirla) [reply src/StreamService/Program.cs]: fixed

File: src/StreamService.Data.Provider/IDataProvider.cs [hunk 9]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace StreamService.Data.Provider
+{
+    public interface IDataProvider
```
Comment (asyagraf) [inline src/StreamService.Data.Provider/IDataProvider.cs]: IDataProvider inherits IBaseDataProvider, don't forget to add IBaseDataProvider methods to StreamServiceDbContext

File: src/StreamService/Program.cs (hunk 3 above)
Comment (asyagraf) [inline src/StreamService/Program.cs]: fix tabs

File: src/StreamService/Startup.cs (hunk 5 above)
Comment (asyagraf) [inline src/StreamService/Startup.cs]: fix tabs

File: src/StreamService/Program.cs [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+using Serilog;
+using System;
+
+namespace LT.DigitalOffice.StreamService
+{
+  public class Program
+    {
+      public static void Main(string[] args)
+      {
+          var configuration = new ConfigurationBuilder()
```
Comment (asyagraf) [inline src/StreamService/Program.cs]: fix tabs

File: src/StreamService/Startup.cs [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+using StackExchange.Redis;
+using LT.DigitalOffice.Kernel.RedisSupport.Configurations;
+using LT.DigitalOffice.StreamService.Models.Dto.Configuration;
+using LT.DigitalOffice.Kernel.RedisSupport.Helpers.Interfaces;
+using LT.DigitalOffice.Kernel.RedisSupport.Helpers;
+using LT.DigitalOffice.Kernel.Helpers;
+
+namespace LT.DigitalOffice.StreamService
+{
+  public class Startup : BaseApiInfo
```
Comment (asyagraf) [inline src/StreamService/Startup.cs:23]: fix tabs

File: src/StreamService.Data.Provider/IDataProvider.cs (hunk 9 above)
Comment (irlirla) [reply src/StreamService.Data.Provider/IDataProvider.cs]: IDataProvider was deleted

File: src/StreamService/Program.cs (hunk 10 above)
Comment (irlirla) [reply src/StreamService/Program.cs]: fixed

File: src/StreamService/Startup.cs (hunk 11 above)
Comment (irlirla) [reply src/StreamService/Startup.cs:23]: fixed

File: Dockerfile [hunk 12]
```diff
@@ -0,0 +1,14 @@
+FROM mcr.microsoft.com/dotnet/sdk:5.0-buster-slim AS build
```
Comment (ndoriomedov) [inline Dockerfile]: 6.0-bullseye-slim

File: src/StreamService.Broker/StreamService.Broker.csproj [hunk 13]
```diff
@@ -0,0 +1,15 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
```
Comment (ndoriomedov) [inline src/StreamService.Broker/StreamService.Broker.csproj]: project gets target framework from Common.Configuration.targets here (line 9), no need in line 4

You can move these properties to Common.Configuration.targets too and remove them from other .csproj files
```
<ImplicitUsings>enable</ImplicitUsings>
<Nullable>enable</Nullable>
```

Comment (ndoriomedov) [reply src/StreamService.Broker/StreamService.Broker.csproj]: by the way, what do these lines mean?
```
<ImplicitUsings>enable</ImplicitUsings>
<Nullable>enable</Nullable>
```

File: src/StreamService.Business/StreamService.Business.csproj [hunk 14]
```diff
@@ -0,0 +1,11 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
```
Comment (ndoriomedov) [inline src/StreamService.Business/StreamService.Business.csproj]: same comment as in StreamService.Broker.csproj

File: src/StreamService.Models.Dto/StreamService.Models.Dto.csproj [hunk 15]
```diff
@@ -0,0 +1,9 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+  </PropertyGroup>
+
+  <Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
```
Comment (ndoriomedov) [inline src/StreamService.Models.Dto/StreamService.Models.Dto.csproj]: same comment as in StreamService.Broker.csproj

File: src/StreamService.Validation/StreamService.Validation.csproj [hunk 16]
```diff
@@ -0,0 +1,11 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+  
+  <Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
```
Comment (ndoriomedov) [inline src/StreamService.Validation/StreamService.Validation.csproj]: same comment as in StreamService.Broker.csproj

File: src/StreamService/Properties/launchSettings.json [hunk 17]
```diff
@@ -0,0 +1,15 @@
+﻿{
+  "profiles": {
+    "StreamService": {
+      "commandName": "Project",
+      "dotnetRunMessages": "true",
+      "launchBrowser": false,
+      "applicationUrl": "http://localhost:5000",
```
Comment (ndoriomedov) [inline src/StreamService/Properties/launchSettings.json]: Do we need this?

File: Dockerfile (hunk 12 above)
Comment (irlirla) [reply Dockerfile]: was fixed in later prs

File: src/StreamService.Broker/StreamService.Broker.csproj (hunk 13 above)
Comment (irlirla) [reply src/StreamService.Broker/StreamService.Broker.csproj]: it was generated by vs and later fixed

File: src/StreamService.Business/StreamService.Business.csproj (hunk 14 above)
Comment (irlirla) [reply src/StreamService.Business/StreamService.Business.csproj]: it was generated by vs and later fixed

File: src/StreamService.Models.Dto/StreamService.Models.Dto.csproj (hunk 15 above)
Comment (irlirla) [reply src/StreamService.Models.Dto/StreamService.Models.Dto.csproj]: it was generated by vs and later fixed

File: src/StreamService.Validation/StreamService.Validation.csproj (hunk 16 above)
Comment (irlirla) [reply src/StreamService.Validation/StreamService.Validation.csproj]: it was generated by vs and later fixed

File: src/StreamService/Properties/launchSettings.json (hunk 17 above)
Comment (irlirla) [reply src/StreamService/Properties/launchSettings.json]: fixed

File: src/StreamService.Models.Dto/Configuration/RabbitMqConfig.cs [hunk 18]
```diff
@@ -0,0 +1,9 @@
+﻿using LT.DigitalOffice.Kernel.BrokerSupport.Configurations;
+
+namespace LT.DigitalOffice.StreamService.Models.Dto.Configuration
+{
+  public class RabbitMqConfig : BaseRabbitMqConfig
+  {
+
```
Comment (Cheephot) [inline src/StreamService.Models.Dto/Configuration/RabbitMqConfig.cs]: remove line

File: src/StreamService/Program.cs [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        string seqApiKey = Environment.GetEnvironmentVariable("seqApiKey");
+        if (string.IsNullOrEmpty(seqApiKey))
+        {
+          seqApiKey = configuration["Serilog:WriteTo:1:Args:apiKey"];
+        }
+
+        Log.Logger = new LoggerConfiguration().ReadFrom
+            .Configuration(configuration)
```
Comment (Cheephot) [inline src/StreamService/Program.cs]: fix tabs

File: src/StreamService/Program.cs [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+            .Enrich.WithProperty("Service", "StreamService")
+            .WriteTo.Seq(
+                serverUrl: seqServerUrl,
```
Comment (Cheephot) [inline src/StreamService/Program.cs]: fix tabs

File: src/StreamService/Program.cs [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+        }
+        finally
+        {
+          Log.CloseAndFlush();
+        }
+      }
+
+      public static IHostBuilder CreateHostBuilder(string[] args) =>
+        Host.CreateDefaultBuilder(args)
+            .UseSerilog()
```
Comment (Cheephot) [inline src/StreamService/Program.cs]: fix tabs

File: src/StreamService/Startup.cs [hunk 22]
```diff
@@ -0,0 +1,207 @@
...
+        GetString("RabbitMqPassword", _rabbitMqConfig.Password, _serviceInfoConfig.Id, "Password"));
+    }
+
+    private void ConfigureMassTransit(IServiceCollection services)
+    {
+      (string username, string password) = GetRabbitMqCredentials();
+
+      services.AddMassTransit(x =>
+      {
+
```
Comment (Cheephot) [inline src/StreamService/Startup.cs]: remove empty line

File: src/StreamService/Startup.cs [hunk 23]
```diff
@@ -0,0 +1,207 @@
...
+        x.UsingRabbitMq((context, cfg) =>
+        {
+          cfg.Host(_rabbitMqConfig.Host, "/", host =>
+          {
+            host.Username(username);
+            host.Password(password);
+          });
+
```
Comment (Cheephot) [inline src/StreamService/Startup.cs]: same

File: src/StreamService/Program.cs (hunk 19 above)
Comment (irlirla) [reply src/StreamService/Program.cs]: fixed

File: src/StreamService/Program.cs (hunk 20 above)
Comment (irlirla) [reply src/StreamService/Program.cs]: fixed

File: src/StreamService/Program.cs (hunk 21 above)
Comment (irlirla) [reply src/StreamService/Program.cs]: fixed

File: src/StreamService/Startup.cs (hunk 22 above)
Comment (irlirla) [reply src/StreamService/Startup.cs]: fixed

File: src/StreamService/Startup.cs (hunk 23 above)
Comment (irlirla) [reply src/StreamService/Startup.cs]: fixed

File: src/StreamService.Models.Dto/Configuration/RabbitMqConfig.cs (hunk 18 above)
Comment (irlirla) [reply src/StreamService.Models.Dto/Configuration/RabbitMqConfig.cs]: fixed

File: Common.Configurations.targets [hunk 24]
```diff
@@ -0,0 +1,40 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$(AssemblyName.Contains('.UnitTests'))">
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup Condition="$(AssemblyName.Contains('.UnitTests')) != 'true'">
+    <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.53" />
+    <PackageReference Include="LT.DigitalOffice.Kernel.BrokerSupport" Version="1.0.1" />
+    <PackageReference Include="LT.DigitalOffice.Kernel.RedisSupport" Version="1.0.1" />
```
Comment (MariyaSin) [inline Common.Configurations.targets]: i do not sure what we will use redis here, could you remove it?

File: src/StreamService/appsettings.json [hunk 25]
```diff
@@ -0,0 +1,47 @@
...
+      "WithThreadId"
+    ],
+    "WriteTo": [
+      {
+        "Name": "Console"
+      },
+      {
+        "Name": "Seq",
+        "Args": {
+          "serverUrl": "http://localhost:9840",
```
Comment (MariyaSin) [inline src/StreamService/appsettings.json]: "http://localhost:5341"

File: Common.Configurations.targets (hunk 24 above)
Comment (irlirla) [reply Common.Configurations.targets]: removed

File: src/StreamService/appsettings.json (hunk 25 above)
Comment (irlirla) [reply src/StreamService/appsettings.json]: fixed

File: src/StreamService/Program.cs [hunk 26]
```diff
@@ -0,0 +1,59 @@
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Serilog;
+using System;
+
+namespace LT.DigitalOffice.StreamService
+{
+  public class Program
+    {
```
Comment (ekaterinaryzhenkova) [inline src/StreamService/Program.cs:12]: fix tabs

File: .editorconfig [hunk 27]
```diff
@@ -0,0 +1,12 @@
+# To learn more about .editorconfig see https://aka.ms/editorconfigdocs
+
```
Comment (ekaterinaryzhenkova) [inline .editorconfig]: are you sure this file look like this? maybe consult with Mariya?

File: src/StreamService/Program.cs (hunk 26 above)
Comment (irlirla) [reply src/StreamService/Program.cs:12]: fixed

**Review by MariyaSin — APPROVED** [review]

File: .editorconfig (hunk 27 above)
Comment (irlirla) [reply .editorconfig]: fixed

File: Common.Configurations.targets [hunk 28]
```diff
@@ -0,0 +1,39 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$(AssemblyName.Contains('.UnitTests'))">
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup Condition="$(AssemblyName.Contains('.UnitTests')) != 'true'">
+    <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.53" />
+    <PackageReference Include="LT.DigitalOffice.Kernel.BrokerSupport" Version="1.0.1" />
+    <PackageReference Include="LT.DigitalOffice.Models" Version="1.0.51" />
```
Comment (ndoriomedov) [inline Common.Configurations.targets]: 1.0.53

File: src/StreamService/appsettings.Development.json [hunk 29]
```diff
@@ -0,0 +1,21 @@
...
+      }
+    }
+  },
+  "ConnectionStrings": {
+  },
+  "RabbitMQ": {
+    "Host": "localhost",
+    "Username": "guest",
+    "Password": "guest"
+  },
```
Comment (ndoriomedov) [inline src/StreamService/appsettings.Development.json]: Add cacheLive for memory cache

File: src/StreamService/appsettings.Production.json [hunk 30]
```diff
@@ -0,0 +1,10 @@
+{
+  "Serilog": {
+  },
+  "RabbitMQ": {
+    "Host": "lt_do_rabbitmq"
+  },
+  "MemoryCache": {
+    "CacheLiveInMinutes": 120
+  }
```
Comment (ndoriomedov) [inline src/StreamService/appsettings.Production.json:6]: Add CacheLive for Redis

Comment (ndoriomedov) [inline src/StreamService/appsettings.Production.json:6]: Add CacheLive for Redis

**Review by ndoriomedov — APPROVED** [review]

File: src/StreamService/appsettings.Production.json (hunk 30 above)
Comment (irlirla) [reply src/StreamService/appsettings.Production.json:6]: Redis was removed from this service for now

Comment (irlirla) [reply src/StreamService/appsettings.Production.json:6]: Redis was removed from this service for now

File: src/StreamService/appsettings.Development.json (hunk 29 above)
Comment (irlirla) [reply src/StreamService/appsettings.Development.json]: appsettings.Production contains it, should it be in Development?

File: Common.Configurations.targets (hunk 28 above)
Comment (irlirla) [reply Common.Configurations.targets]: fixed

**Review by grinakate — APPROVED** [review]

File: .dockerignore [hunk 31]
```diff
@@ -0,0 +1,2 @@
+bin/
```
Comment (VladSpitsin) [inline .dockerignore:1]: are you sure file should look like this?

**Review by VladSpitsin — APPROVED** [review]

**Pull request merged**